README.md
# Factorio-Leighzers-Uranium-Logistics
This is a repository for the factorio mod Leighzer's Uranium Logistics.

Source/changelog.txt
---------------------------------------------------------------------------------------------------
Version: 0.1.3
  Changes:
    - Add advanced-electronics-2 as prereq to uranium-logistics.
---------------------------------------------------------------------------------------------------
Version: 0.1.2
  Bugfixes:
    - Fixed misspelled file name for normal res uranium-underground-belt remnant graphic.
---------------------------------------------------------------------------------------------------
Version: 0.1.1
  Bugfixes:
    - Removed accidental next_upgrade for uranium-underground-belt.
---------------------------------------------------------------------------------------------------
Version: 0.1.0
  Changes:
    - Initial release.


Source/data.lua
if not leighzermods then --generic mod variable to store information that may be needed later
    leighzermods = {}
end

if not leighzermods.leighzeruraniumlogistics then
    leighzermods.leighzeruraniumlogistics = {}

    leighzermods.leighzeruraniumlogistics.beltSpeed = 0.03125 * 5 -- 5x regular belt speed 75/s 4500/min
    leighzermods.leighzeruraniumlogistics.undergroundBeltMaxLength = 11 -- following pattern 5, 7, 9, => 11 for reg, fast, express => uranium
end

--load in items, recipes, entities, and remnant entities
require("prototypes.technology")
require("prototypes.uranium-splitter")
require("prototypes.uranium-transport-belt")
require("prototypes.uranium-underground-belt")

-- update express logistics to be upgradeable to our new belts
if data.raw.splitter["express-splitter"] then
    data.raw.splitter["express-splitter"].next_upgrade = "uranium-splitter"
end

if data.raw["transport-belt"]["express-transport-belt"] then
    data.raw["transport-belt"]["express-transport-belt"].next_upgrade = "uranium-transport-belt"
end

if data.raw["underground-belt"]["express-underground-belt"] then
    data.raw["underground-belt"]["express-underground-belt"].next_upgrade = "uranium-underground-belt"
end

Source/prototypes/technology.lua
local uraniumLogisticsTech = 
  {
    type = "technology",
    name = "uranium-logistics",
    icon_size = 128,
    icon = "__leighzeruraniumlogistics__/graphics/technology/uranium-logistics.png",
    effects =
    {
      {
        type = "unlock-recipe",
        recipe = "uranium-transport-belt"
      },
      {
        type = "unlock-recipe",
        recipe = "uranium-underground-belt"
      },
      {
        type = "unlock-recipe",
        recipe = "uranium-splitter"
      }
    },
    prerequisites = {"kovarex-enrichment-process"},
    unit =
    {
      count = 1500,
      ingredients =
      {
        {"automation-science-pack", 1},
        {"logistic-science-pack", 1},
        {"chemical-science-pack", 1},
        {"production-science-pack", 1}
      },
      time = 30
    },
    order = "e-p-b-d"
  }

if data.raw.technology["logistics-3"] then
  table.insert(uraniumLogisticsTech.prerequisites, "logistics-3")
end

if data.raw.technology["advanced-electronics-2"] then
  table.insert(uraniumLogisticsTech.prerequisites, "advanced-electronics-2")
end

data:extend({
    uraniumLogisticsTech
})

Source/prototypes/uranium-splitter.lua
require("prototypes.uranium-transport-belt-pictures")

local hit_effects = require ("__base__.prototypes.entity.demo-hit-effects")
local sounds = require("__base__.prototypes.entity.demo-sounds")

data:extend({  

  {
    type = "item",
    name = "uranium-splitter",
    icon = "__leighzeruraniumlogistics__/graphics/icons/uranium-splitter.png",
    icon_size = 64, icon_mipmaps = 4,
    subgroup = "belt",
    order = "c[splitter]-d[uranium-splitter]",
    place_result = "uranium-splitter",
    stack_size = 50
  },

  {
    type = "recipe",
    name = "uranium-splitter",
    category = "crafting-with-fluid",
    enabled = false,
    energy_required = 1,
    ingredients =
    {
      {"processing-unit", 10},
      {"express-splitter", 1},
      {name="sulfuric-acid",amount=200,type="fluid"},
      {"uranium-235",3},
      {"steel-plate",10}
    },
    result = "uranium-splitter"
  },

  {
      type = "splitter",
      name = "uranium-splitter",
      icon = "__leighzeruraniumlogistics__/graphics/icons/uranium-splitter.png",
      icon_size = 64, icon_mipmaps = 4,
      flags = {"placeable-neutral", "player-creation"},
      minable = {mining_time = 0.1, result = "uranium-splitter"},
      max_health = 200,
      corpse = "uranium-splitter-remnants",
      dying_explosion = "medium-explosion",--"splitter-explosion",
      resistances =
      {
        {
          type = "fire",
          percent = 60
        }
      },
      collision_box = {{-0.9, -0.4}, {0.9, 0.4}},
      selection_box = {{-0.9, -0.5}, {0.9, 0.5}},
      damaged_trigger_effect = hit_effects.entity(),
      animation_speed_coefficient = 32,
      structure_animation_speed_coefficient = 0.7,
      structure_animation_movement_cooldown = 10,
      fast_replaceable_group = "transport-belt",      
      speed = leighzermods.leighzeruraniumlogistics.beltSpeed,
      belt_animation_set = uranium_basic_belt_animation_set,
      working_sound = sounds.splitter,
      structure =
      {
        north =
        {
          filename = "__leighzeruraniumlogistics__/graphics/entity/uranium-splitter/uranium-splitter-north.png",
          frame_count = 32,
          line_length = 8,
          priority = "extra-high",
          width = 82,
          height = 36,
          shift = util.by_pixel(6, 0),
          hr_version =
          {
            filename = "__leighzeruraniumlogistics__/graphics/entity/uranium-splitter/hr-uranium-splitter-north.png",
            frame_count = 32,
            line_length = 8,
            priority = "extra-high",
            width = 160,
            height = 70,
            shift = util.by_pixel(7, 0),
            scale = 0.5
          }
        },
        east =
        {
          filename = "__leighzeruraniumlogistics__/graphics/entity/uranium-splitter/uranium-splitter-east.png",
          frame_count = 32,
          line_length = 8,
          priority = "extra-high",
          width = 46,
          height = 44,
          shift = util.by_pixel(4, 12),
          hr_version =
          {
            filename = "__leighzeruraniumlogistics__/graphics/entity/uranium-splitter/hr-uranium-splitter-east.png",
            frame_count = 32,
            line_length = 8,
            priority = "extra-high",
            width = 90,
            height = 84,
            shift = util.by_pixel(4, 13),
            scale = 0.5
          }
        },
        south =
        {
          filename = "__leighzeruraniumlogistics__/graphics/entity/uranium-splitter/uranium-splitter-south.png",
          frame_count = 32,
          line_length = 8,
          priority = "extra-high",
          width = 82,
          height = 32,
          shift = util.by_pixel(4, 0),
          hr_version =
          {
            filename = "__leighzeruraniumlogistics__/graphics/entity/uranium-splitter/hr-uranium-splitter-south.png",
            frame_count = 32,
            line_length = 8,
            priority = "extra-high",
            width = 164,
            height = 64,
            shift = util.by_pixel(4, 0),
            scale = 0.5
          }
        },
        west =
        {
          filename = "__leighzeruraniumlogistics__/graphics/entity/uranium-splitter/uranium-splitter-west.png",
          frame_count = 32,
          line_length = 8,
          priority = "extra-high",
          width = 46,
          height = 44,
          shift = util.by_pixel(6, 12),
          hr_version =
          {
            filename = "__leighzeruraniumlogistics__/graphics/entity/uranium-splitter/hr-uranium-splitter-west.png",
            frame_count = 32,
            line_length = 8,
            priority = "extra-high",
            width = 90,
            height = 86,
            shift = util.by_pixel(6, 12),
            scale = 0.5
          }
        }
      },
      structure_patch =
      {
        north = util.empty_sprite(),
        east =
        {
          filename = "__leighzeruraniumlogistics__/graphics/entity/uranium-splitter/uranium-splitter-east-top_patch.png",
          frame_count = 32,
          line_length = 8,
          priority = "extra-high",
          width = 46,
          height = 52,
          shift = util.by_pixel(4, -20),
          hr_version =
          {
            filename = "__leighzeruraniumlogistics__/graphics/entity/uranium-splitter/hr-uranium-splitter-east-top_patch.png",
            frame_count = 32,
            line_length = 8,
            priority = "extra-high",
            width = 90,
            height = 104,
            shift = util.by_pixel(4, -20),
            scale = 0.5
          }
        },
        south = util.empty_sprite(),
        west =
        {
          filename = "__leighzeruraniumlogistics__/graphics/entity/uranium-splitter/uranium-splitter-west-top_patch.png",
          frame_count = 32,
          line_length = 8,
          priority = "extra-high",
          width = 46,
          height = 48,
          shift = util.by_pixel(6, -18),
          hr_version =
          {
            filename = "__leighzeruraniumlogistics__/graphics/entity/uranium-splitter/hr-uranium-splitter-west-top_patch.png",
            frame_count = 32,
            line_length = 8,
            priority = "extra-high",
            width = 90,
            height = 96,
            shift = util.by_pixel(6, -18),
            scale = 0.5
          }
        }
      }
    },
    {
      type = "corpse",
      name = "uranium-splitter-remnants",
      icon = "__leighzeruraniumlogistics__/graphics/icons/uranium-splitter.png",
      icon_size = 64, icon_mipmaps = 4,
      flags = {"placeable-neutral", "not-on-map"},
      subgroup = "belt-remnants",
      order = "a-j-0",
      selection_box = {{-0.9, -0.5}, {0.9, 0.5}},
      tile_width = 2,
      tile_height = 1,
      selectable_in_game = false,
      time_before_removed = 60 * 60 * 15, -- 15 minutes
      final_render_layer = "remnants",
      remove_on_tile_placement = false,
      animation =
      {
        filename = "__leighzeruraniumlogistics__/graphics/entity/uranium-splitter/remnants/uranium-splitter-remnants.png",
        line_length = 1,
        width = 96,
        height = 96,
        frame_count = 1,
        variation_count = 1,
        axially_symmetrical = false,
        direction_count = 4,
        shift = util.by_pixel(4, 1.5),
        hr_version =
        {
          filename = "__leighzeruraniumlogistics__/graphics/entity/uranium-splitter/remnants/hr-uranium-splitter-remnants.png",
          line_length = 1,
          width = 190,
          height = 190,
          frame_count = 1,
          variation_count = 1,
          axially_symmetrical = false,
          direction_count = 4,
          shift = util.by_pixel(3.5, 1.5),
          scale = 0.5,
        },
      }
    },
})

Source/prototypes/uranium-transport-belt.lua
require("prototypes.uranium-transport-belt-pictures")

local hit_effects = require ("__base__.prototypes.entity.demo-hit-effects")
local sounds = require("__base__.prototypes.entity.demo-sounds")

data:extend({ 
  {
    type = "item",
    name = "uranium-transport-belt",
    icon = "__leighzeruraniumlogistics__/graphics/icons/uranium-transport-belt.png",
    icon_size = 64, icon_mipmaps = 4,
    subgroup = "belt",
    order = "a[transport-belt]-d[uranium-transport-belt]",
    place_result = "uranium-transport-belt",
    stack_size = 100
  },

  {
    type = "recipe",
    name = "uranium-transport-belt",
    category = "crafting-with-fluid",
    enabled = false,
    ingredients =
    {
      {"express-transport-belt", 1},
      {"uranium-235", 1},
      {name="sulfuric-acid", amount=50, type="fluid"},
      {"steel-plate",20}
    },
    result = "uranium-transport-belt"    
  },

  {
      type = "transport-belt",
      name = "uranium-transport-belt",
      icon = "__leighzeruraniumlogistics__/graphics/icons/uranium-transport-belt.png",
      icon_size = 64, icon_mipmaps = 4,
      flags = {"placeable-neutral", "player-creation"},
      minable = {mining_time = 0.1, result = "uranium-transport-belt"},
      max_health = 180,
      corpse = "uranium-transport-belt-remnants",
      dying_explosion = "medium-explosion",
      resistances =
      {
        {
          type = "fire",
          percent = 50
        }
      },
      collision_box = {{-0.4, -0.4}, {0.4, 0.4}},
      selection_box = {{-0.5, -0.5}, {0.5, 0.5}},
      damaged_trigger_effect = hit_effects.entity(),
      working_sound =
      {
        sound =
        {
          filename = "__base__/sound/transport-belt.ogg",
          volume = 0.3
        },
        persistent = true
      },
      animation_speed_coefficient = 32,
      belt_animation_set = uranium_basic_belt_animation_set,
      fast_replaceable_group = "transport-belt",      
      speed = leighzermods.leighzeruraniumlogistics.beltSpeed,
      connector_frame_sprites = transport_belt_connector_frame_sprites,
      circuit_wire_connection_points = circuit_connector_definitions["belt"].points,
      circuit_connector_sprites = circuit_connector_definitions["belt"].sprites,
      circuit_wire_max_distance = transport_belt_circuit_wire_max_distance
  },
  {
    type = "corpse",
    name = "uranium-transport-belt-remnants",
    icon = "__leighzeruraniumlogistics__/graphics/icons/uranium-transport-belt.png",
    icon_size = 64, icon_mipmaps = 4,
    flags = {"placeable-neutral", "not-on-map"},
    subgroup = "belt-remnants",
    order = "a-d-0",
    selection_box = {{-0.5, -0.5}, {0.5, 0.5}},
    tile_width = 1,
    tile_height = 1,
    selectable_in_game = false,
    time_before_removed = 60 * 60 * 15, -- 15 minutes
    final_render_layer = "remnants",
    animation =  make_rotated_animation_variations_from_sheet (2,
    {
      filename = "__leighzeruraniumlogistics__/graphics/entity/uranium-transport-belt/remnants/uranium-transport-belt-remnants.png",
      line_length = 1,
      width = 54,
      height = 52,
      frame_count = 1,
      variation_count = 1,
      axially_symmetrical = false,
      direction_count = 4,
      shift = util.by_pixel(1, 0),
      hr_version =
      {
        filename = "__leighzeruraniumlogistics__/graphics/entity/uranium-transport-belt/remnants/hr-uranium-transport-belt-remnants.png",
        line_length = 1,
        width = 106,
        height = 102,
        frame_count = 1,
        variation_count = 1,
        axially_symmetrical = false,
        direction_count = 4,
        shift = util.by_pixel(1, -0.5),
        scale = 0.5,
      },
    })
  }
})


Source/prototypes/uranium-underground-belt.lua
require("prototypes.uranium-transport-belt-pictures")

local hit_effects = require ("__base__.prototypes.entity.demo-hit-effects")
local sounds = require("__base__.prototypes.entity.demo-sounds")

data:extend({
  
  {
    type = "item",
    name = "uranium-underground-belt",
    icon = "__leighzeruraniumlogistics__/graphics/icons/uranium-underground-belt.png",    
    icon_size = 64, icon_mipmaps = 4,
    subgroup = "belt",
    order = "b[underground-belt]-d[uranium-underground-belt]",
    place_result = "uranium-underground-belt",
    stack_size = 50
  },

  {
    type = "recipe",
    name = "uranium-underground-belt",
    category = "crafting-with-fluid",
    enabled = false,
    energy_required = 1,
    ingredients =
    {
      {"express-underground-belt",2},
      {"steel-plate",160},
      {"uranium-235",10},
      {name="sulfuric-acid",amount=100,type="fluid"}
    },
    result_count = 2,
    result = "uranium-underground-belt"
  },

  {
      type = "underground-belt",
      name = "uranium-underground-belt",
      icon = "__leighzeruraniumlogistics__/graphics/icons/uranium-underground-belt.png",    
      icon_size = 64, icon_mipmaps = 4,
      flags = {"placeable-neutral", "player-creation"},
      minable = {mining_time = 0.1, result = "uranium-underground-belt"},
      max_health = 180,
      corpse = "uranium-underground-belt-remnants",
      dying_explosion = "medium-explosion",
      max_distance = leighzermods.leighzeruraniumlogistics.undergroundBeltMaxLength,
      underground_sprite =
      {
        filename = "__core__/graphics/arrows/underground-lines.png",
        priority = "high",
        width = 64,
        height = 64,
        x = 64,
        scale = 0.5
      },
      underground_remove_belts_sprite =
      {
        filename = "__core__/graphics/arrows/underground-lines-remove.png",
        priority = "high",
        width = 64,
        height = 64,
        x = 64,
        scale = 0.5
      },
      resistances =
      {
        {
          type = "fire",
          percent = 60
        },
        {
          type = "impact",
          percent = 30
        }
      },
      collision_box = {{-0.4, -0.4}, {0.4, 0.4}},
      selection_box = {{-0.5, -0.5}, {0.5, 0.5}},
      damaged_trigger_effect = hit_effects.entity(),
      animation_speed_coefficient = 32,
      belt_animation_set = uranium_basic_belt_animation_set,
      fast_replaceable_group = "transport-belt",      
      speed = leighzermods.leighzeruraniumlogistics.beltSpeed,
      structure =
      {
        direction_in =
        {
          sheet =
          {
            filename = "__leighzeruraniumlogistics__/graphics/entity/uranium-underground-belt/uranium-underground-belt-structure.png",
            priority = "extra-high",
            width = 96,
            height = 96,
            y = 96,
            hr_version =
            {
              filename = "__leighzeruraniumlogistics__/graphics/entity/uranium-underground-belt/hr-uranium-underground-belt-structure.png",
              priority = "extra-high",
              width = 192,
              height =192,
              y = 192,
              scale = 0.5
            }
          }
        },
        direction_out =
        {
          sheet =
          {
            filename = "__leighzeruraniumlogistics__/graphics/entity/uranium-underground-belt/uranium-underground-belt-structure.png",
            priority = "extra-high",
            width = 96,
            height = 96,
            hr_version =
            {
              filename = "__leighzeruraniumlogistics__/graphics/entity/uranium-underground-belt/hr-uranium-underground-belt-structure.png",
              priority = "extra-high",
              width = 192,
              height = 192,
              scale = 0.5
            }
  
          }
  
        },
        direction_in_side_loading =
        {
          sheet =
          {
            filename = "__leighzeruraniumlogistics__/graphics/entity/uranium-underground-belt/uranium-underground-belt-structure.png",
            priority = "extra-high",
            width = 96,
            height = 96,
            y = 96*3,
            hr_version =
            {
              filename = "__leighzeruraniumlogistics__/graphics/entity/uranium-underground-belt/hr-uranium-underground-belt-structure.png",
              priority = "extra-high",
              width = 192,
              height = 192,
              y = 192*3,
              scale = 0.5
            }
          }
        },
        direction_out_side_loading =
        {
          sheet =
          {
            filename = "__leighzeruraniumlogistics__/graphics/entity/uranium-underground-belt/uranium-underground-belt-structure.png",
            priority = "extra-high",
            width = 96,
            height = 96,
            y = 96*2,
            hr_version =
            {
              filename = "__leighzeruraniumlogistics__/graphics/entity/uranium-underground-belt/hr-uranium-underground-belt-structure.png",
              priority = "extra-high",
              width = 192,
              height = 192,
              y= 192*2,
              scale = 0.5
            },
  
          }
  
        },
        back_patch =
        {
          sheet =
          {
            filename = "__base__/graphics/entity/underground-belt/underground-belt-structure-back-patch.png",
            priority = "extra-high",
            width = 96,
            height = 96,
            hr_version =
            {
              filename = "__base__/graphics/entity/underground-belt/hr-underground-belt-structure-back-patch.png",
              priority = "extra-high",
              width = 192,
              height = 192,
              scale = 0.5
            }
  
          }
        },
        front_patch =
        {
          sheet =
          {
            filename = "__base__/graphics/entity/underground-belt/underground-belt-structure-front-patch.png",
            priority = "extra-high",
            width = 96,
            height = 96,
            hr_version =
            {
              filename = "__base__/graphics/entity/underground-belt/hr-underground-belt-structure-front-patch.png",
              priority = "extra-high",
              width = 192,
              height = 192,
              scale = 0.5
            }
  
          }
        }
      }
    },
    {
      type = "corpse",
      name = "uranium-underground-belt-remnants",
      icon = "__leighzeruraniumlogistics__/graphics/icons/uranium-underground-belt.png",
      icon_size = 64, icon_mipmaps = 4,
      flags = {"placeable-neutral", "not-on-map", "building-direction-8-way"},
      subgroup = "belt-remnants",
      order="a-g-0",
      selection_box = {{-0.5, -0.5}, {0.5, 0.5}},
      tile_width = 1,
      tile_height = 1,
      selectable_in_game = false,
      time_before_removed = 60 * 60 * 15, -- 15 minutes
      final_render_layer = "remnants",
      remove_on_tile_placement = false,
      animation =
      {
        filename = "__leighzeruraniumlogistics__/graphics/entity/uranium-underground-belt/remnants/uranium-underground-belt-remnants.png",
        line_length = 1,
        width = 78,
        height =72,
        frame_count = 1,
        variation_count = 1,
        axially_symmetrical = false,
        direction_count = 8,
        shift = util.by_pixel(10, 3),
        hr_version =
        {
          filename = "__leighzeruraniumlogistics__/graphics/entity/uranium-underground-belt/remnants/hr-uranium-underground-belt-remnants.png",
          line_length = 1,
          width = 156,
          height = 144,
          frame_count = 1,
          variation_count = 1,
          axially_symmetrical = false,
          direction_count = 8,
          shift = util.by_pixel(10.5, 3),
          scale = 0.5,
        },
      }
    },
})